ui/ducks/bridge/selectors.ts

```diff
@@ -130,28 +130,51 @@ export const getToToken = (
 export const getFromAmount = (state: BridgeAppState): string | null =>
   state.bridge.fromTokenInputValue;
 
-export const getBridgeQuotes = (state: BridgeAppState) => {
-  return {
-    quotes: state.metamask.bridgeState.quotes,
-    quotesLastFetchedMs: state.metamask.bridgeState.quotesLastFetched,
-    isLoading:
-      state.metamask.bridgeState.quotesLoadingStatus === RequestStatus.LOADING,
-  };
+export const getQuoteRequest = (state: BridgeAppState) => {
+  const { quoteRequest } = state.metamask.bridgeState;
+  return quoteRequest;
 };
 
+export const getBridgeQuotesConfig = (state: BridgeAppState) =>
+  state.metamask.bridgeState?.bridgeFeatureFlags[
+    BridgeFeatureFlagsKey.EXTENSION_CONFIG
+  ] ?? {};
+
+export const getBridgeQuotes = createSelector(
+  (state: BridgeAppState) => state.metamask.bridgeState.quotes,
+  (state: BridgeAppState) => state.metamask.bridgeState.quotesLastFetched,
+  (state: BridgeAppState) =>
+    state.metamask.bridgeState.quotesLoadingStatus === RequestStatus.LOADING,
+  (state: BridgeAppState) => state.metamask.bridgeState.quotesRefreshCount,
+  getBridgeQuotesConfig,
+  getQuoteRequest,
+  (
+    quotes,
+    quotesLastFetchedMs,
+    isLoading,
+    quotesRefreshCount,
+    { maxRefreshCount },
+    { insufficientBal },
+  ) => {
+    return {
+      quotes,
+      quotesLastFetchedMs,
+      isLoading,
+      quotesRefreshCount,
+      isQuoteGoingToRefresh: insufficientBal
+        ? false
+        : quotesRefreshCount < maxRefreshCount,
+    };
+  },
+);
+
 export const getRecommendedQuote = createSelector(
   getBridgeQuotes,
   ({ quotes }) => {
```

Comment:
Is this still a TODO?

Reply:
The logic for this is in a [separate PR](https://github.com/MetaMask/metamask-extension/pull/27731/commits). Not including it in this one to unblock yours